## **Integrantes**

Juan Diego Correa Noreña,
Yessica Bastidas Villada, 
Santiago Bustos Pianda,
Juan Carlos Gomez Ramirez.

<p><a name="s3"></a></p>

# **1. Carga de datos**

In [209]:
# Importar paquetes
import pandas as pd
import numpy as np
import math as ma
import matplotlib as plt
import seaborn as sns
from datetime import date
import plotly.express as px
from plotly.subplots import make_subplots
from google.colab import files

In [210]:
# Conectar con Drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [211]:
# Cargar base de datos 2018
df_2018 = pd.read_csv('/content/drive/MyDrive/ANALÍTICA/Trabajo final/Emergencias_2018_1.csv',sep=';',index_col= False)
df_2018.head()

FECHA  DEPARTAMENTO   MUNICIPIO     TIPO_DE_EVENTO  DIVIPOLA  \
0  1/01/2018        BOYACA    GUATEQUE  INCENDIO FORESTAL   15322.0   
1  1/01/2018  CUNDINAMARCA      GUASCA      DESLIZAMIENTO   25322.0   
2  1/01/2018       QUINDIO  MONTENEGRO      DESLIZAMIENTO   63470.0   
3  1/02/2018     ANTIOQUIA       ANORI      DESLIZAMIENTO    5040.0   
4  1/02/2018         CHOCO       LLORO  ACCIDENTE FLUVIAL   27413.0   

   FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
0           0        0              0          0         0   
1           0        0              0          0         0   
2           0        1              0          1         0   
3           0        0              0         15         3   
4           2        0              0          2         0   

   VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS  
0                     0                    0  
1                     0                    0  
2                     0                    0  
3                     1                    2  
4                     0                    0

In [212]:
# Cargar base de datos 2019
df_2019 = pd.read_csv('/content/drive/MyDrive/ANALÍTICA/Trabajo final/Emergencias_2019_1.csv',sep=';',index_col= False)
df_2019.head()

FECHA DEPARTAMENTO     MUNICIPIO                 TIPO_DE_EVENTO  \
0  1/01/2019    ANTIOQUIA  SAN JERONIMO           INCENDIO ESTRUCTURAL   
1  1/01/2019    RISARALDA      MARSELLA                      INMERSION   
2  1/01/2019    ANTIOQUIA          ANZA  INCENDIO DE COBERTURA VEGETAL   
3  1/01/2019    ANTIOQUIA    MONTEBELLO  INCENDIO DE COBERTURA VEGETAL   
4  1/01/2019    ANTIOQUIA      CAUCASIA           INCENDIO ESTRUCTURAL   

   DIVIPOLA  FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
0      5656           0        0              0          5         1   
1     66440           1        0              0          1         0   
2      5044           0        0              0          0         0   
3      5467           0        0              0          0         0   
4      5154           0        1              0          1         1   

   VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS  
0                     1                    0  
1                     0                    0  
2                     0                    0  
3                     0                    0  
4                     1                    0

In [213]:
# Cargar base de datos 2020
df_2020 = pd.read_csv('/content/drive/MyDrive/ANALÍTICA/Trabajo final/Emergencias_2020_1.csv',sep=';',index_col= False)
df_2020.head()

FECHA DEPARTAMENTO            MUNICIPIO      TIPO_DE_EVENTO  DIVIPOLA  \
0  1/01/2020    RISARALDA  SANTA ROSA DE CABAL  MOVIMIENTO EN MASA     66682   
1  1/01/2020      QUINDIO           MONTENEGRO  MOVIMIENTO EN MASA     63470   
2  1/01/2020        CAUCA                SUCRE  MOVIMIENTO EN MASA     19785   
3  1/01/2020       BOYACA             MOTAVITA              SEQUIA     15476   
4  1/01/2020       BOYACA                PAIPA              SEQUIA     15516   

   FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
0         0.0        0              0         12         3   
1         0.0        0              0          0         0   
2         0.0        0              0         15         3   
3         0.0        0              0          0         0   
4         0.0        0              0          0         0   

   VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS  
0                     2                    1  
1                     0                    0  
2                     0                    3  
3                     0                    0  
4                     0                    0

In [214]:
#se concatenaron las 3 bases ya que tienen las mismas columnas
Basedatos=pd.concat([df_2018,df_2019,df_2020])

In [215]:
Basedatos

FECHA  DEPARTAMENTO            MUNICIPIO  \
0       1/01/2018        BOYACA             GUATEQUE   
1       1/01/2018  CUNDINAMARCA               GUASCA   
2       1/01/2018       QUINDIO           MONTENEGRO   
3       1/02/2018     ANTIOQUIA                ANORI   
4       1/02/2018         CHOCO                LLORO   
...           ...           ...                  ...   
3806  12/30/2020          CAUCA               BALBOA   
3807  12/31/2020      RISARALDA  SANTA ROSA DE CABAL   
3808  12/31/2020          CAUCA              POPAYAN   
3809  12/31/2020      MAGDALENA        NUEVA GRANADA   
3810  12/31/2020          CAUCA              POPAYAN   

                      TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  HERIDOS  \
0                  INCENDIO FORESTAL   15322.0         0.0        0   
1                      DESLIZAMIENTO   25322.0         0.0        0   
2                      DESLIZAMIENTO   63470.0         0.0        1   
3                      DESLIZAMIENTO    5040.0         0.0        0   
4                  ACCIDENTE FLUVIAL   27413.0         2.0        0   
...                              ...       ...         ...      ...   
3806              MOVIMIENTO EN MASA   19075.0         0.0        0   
3807  ACCIDENTE TRANSPORTE TERRESTRE   66682.0         0.0        1   
3808                      INUNDACION   19001.0         0.0        0   
3809            INCENDIO ESTRUCTURAL   47460.0         0.0        0   
3810                      INUNDACION   19001.0         0.0        0   

      DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
0                 0          0         0                     0   
1                 0          0         0                     0   
2                 0          1         0                     0   
3                 0         15         3                     1   
4                 0          2         0                     0   
...             ...        ...       ...                   ...   
3806              0         15         3                     0   
3807              0          1         0                     0   
3808              0          0         0                     0   
3809              0          0         0                     0   
3810              0         20         5                     0   

      VIVIENDAS_AVERIADAS  
0                       0  
1                       0  
2                       0  
3                       2  
4                       0  
...                   ...  
3806                    3  
3807                    0  
3808                    0  
3809                    0  
3810                    5  

[11743 rows x 12 columns]

In [216]:
# Cargar base de datos de coordenadas
pyp = pd.read_csv('/content/drive/MyDrive/ANALÍTICA/Trabajo final/DIV.csv',sep=';',index_col= False)
pyp.head(3)

DIVIPOLA   LATITUD   LONGITUD
0      5001  6.257590 -75.611031
1      5002  5.803728 -75.438474
2      5004  6.627569 -76.085978

In [217]:
#Se hace un merge para unir la base de datos principal con la de coordenadas por medio de los codigos de DIVIPOLA
Basedatos= Basedatos.merge(pyp, on="DIVIPOLA", how="inner")
Basedatos

FECHA DEPARTAMENTO            MUNICIPIO       TIPO_DE_EVENTO  \
0        1/01/2018       BOYACA             GUATEQUE    INCENDIO FORESTAL   
1      01/28/2018        BOYACA             GUATEQUE    INCENDIO FORESTAL   
2        2/09/2018       BOYACA             GUATEQUE    INCENDIO FORESTAL   
3        2/09/2018       BOYACA             GUATEQUE    INCENDIO FORESTAL   
4      02/26/2018        BOYACA             GUATEQUE    INCENDIO FORESTAL   
...            ...          ...                  ...                  ...   
11687  09/25/2020     SANTANDER              CHIPATA     CRECIENTE SUBITA   
11688  09/29/2020       CAQUETA  SAN JOSE DEL FRAGUA             VENDAVAL   
11689  10/20/2020     ANTIOQUIA              YOLOMBO  COLAPSO ESTRUCTURAL   
11690  10/24/2020     ATLANTICO      PUERTO COLOMBIA             VENDAVAL   
11691  11/14/2020     ATLANTICO      PUERTO COLOMBIA   MOVIMIENTO EN MASA   

       DIVIPOLA  FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
0       15322.0         0.0        0              0          0         0   
1       15322.0         0.0        0              0          0         0   
2       15322.0         0.0        0              0          0         0   
3       15322.0         0.0        0              0          0         0   
4       15322.0         0.0        0              0          0         0   
...         ...         ...      ...            ...        ...       ...   
11687   68179.0         0.0        0              0          0         0   
11688   18610.0         0.0        0              0        280        70   
11689    5890.0         1.0        1              0          2         0   
11690    8573.0         0.0        1              0         25         6   
11691    8573.0         0.0        0              0         15         3   

       VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS    LATITUD   LONGITUD  
0                         0                    0   5.014508 -73.488124  
1                         0                    0   5.014508 -73.488124  
2                         0                    0   5.014508 -73.488124  
3                         0                    0   5.014508 -73.488124  
4                         0                    0   5.014508 -73.488124  
...                     ...                  ...        ...        ...  
11687                     0                    0   6.067945 -73.632391  
11688                     0                   70   1.304702 -76.110006  
11689                     1                    0   6.644428 -75.029903  
11690                     1                    5  11.004423 -74.907406  
11691                     0                    3  11.004423 -74.907406  

[11692 rows x 14 columns]

<p><a name="s3"></a></p>

# **2. Depuración: Homologación de categorías, tipos de datos**

In [218]:
#Cambiar formato de la fecha
Basedatos['FECHA'] = pd.to_datetime(Basedatos['FECHA']) 
Basedatos.dtypes 
#La base de datos tiene 1 variable en formato fecha, 3 variables object, 4 variables float y 6 variables enteras

FECHA                   datetime64[ns]
DEPARTAMENTO                    object
MUNICIPIO                       object
TIPO_DE_EVENTO                  object
DIVIPOLA                       float64
FALLECIDOS                     float64
HERIDOS                          int64
DESAPARECIDOS                    int64
AFECTADOS                        int64
FAMILIAS                         int64
VIVIENDAS_DESTRUIDAS             int64
VIVIENDAS_AVERIADAS              int64
LATITUD                        float64
LONGITUD                       float64
dtype: object

###**Revisión de las categorias de cada variable tipo object**

In [221]:
# Se encuentra que hay 34 departamentos, sabiendo que Colombia tiene 32, se procedera a identificar cuales son
print(Basedatos['DEPARTAMENTO'].unique()) 
print(len(Basedatos['DEPARTAMENTO'].unique())) # 34 departamentos

['BOYACA' 'CUNDINAMARCA' 'QUINDIO' 'ANTIOQUIA' 'CHOCO' 'LA GUAJIRA' 'META'
 'CALDAS' 'CESAR' 'RISARALDA' 'TOLIMA' 'SANTANDER' 'SUCRE'
 'VALLE DEL CAUCA' 'BOGOTA, D.C.' 'BOLIVAR' 'NARIÑO' 'NORTE DE SANTANDER'
 'VICHADA' 'CASANARE' 'PUTUMAYO' 'ATLANTICO' 'CAQUETA' 'CAUCA' 'GUAINIA'
 'MAGDALENA' 'HUILA' 'VAUPES' 'GUAVIARE' 'Guaviare' 'CORDOBA' 'ARAUCA'
 'SAN ANDRES' 'AMAZONAS']
34


In [223]:
# 1. Se encuentra que Guaviare esta duplicado
# 2. Bogotá aparece como departamento cuando deberia ser parte de Cundinamarca

# 1. Se remplaza 'Guaviare' por 'GUAVIARE'
Basedatos["DEPARTAMENTO"]= Basedatos["DEPARTAMENTO"].replace("Guaviare", "GUAVIARE")
# 2. Se remplaza 'BOGOTA, D.C.' por 'CUNDINAMARCA'
Basedatos["DEPARTAMENTO"]= Basedatos["DEPARTAMENTO"].replace("BOGOTA, D.C.", "CUNDINAMARCA")

In [224]:
# Se confirma que quedaron 32 departamentos
print(Basedatos["DEPARTAMENTO"].unique())
print(len(Basedatos['DEPARTAMENTO'].unique()))

['BOYACA' 'CUNDINAMARCA' 'QUINDIO' 'ANTIOQUIA' 'CHOCO' 'LA GUAJIRA' 'META'
 'CALDAS' 'CESAR' 'RISARALDA' 'TOLIMA' 'SANTANDER' 'SUCRE'
 'VALLE DEL CAUCA' 'BOLIVAR' 'NARIÑO' 'NORTE DE SANTANDER' 'VICHADA'
 'CASANARE' 'PUTUMAYO' 'ATLANTICO' 'CAQUETA' 'CAUCA' 'GUAINIA' 'MAGDALENA'
 'HUILA' 'VAUPES' 'GUAVIARE' 'CORDOBA' 'ARAUCA' 'SAN ANDRES' 'AMAZONAS']
32


In [228]:
# Se verifica que los municipios esten bien escritos
Basedatos["MUNICIPIO"].unique()

array(['GUATEQUE', 'GUASCA', 'MONTENEGRO', 'ANORI', 'LLORO', 'UBALA',
       'ALBANIA', 'CUBARRAL', 'LA URIBE', 'PUERTO BERRIO', 'SUPIA',
       'CURUMANI', 'BARRANCAS', 'PUEBLO RICO', 'ORTEGA', 'ROVIRA',
       'COCORNA', 'QUINCHIA', 'PUERTO WILCHES', 'CHIQUIZA', 'AGUACHICA',
       'PUERTO GAITAN', 'COVEÑAS', 'VIOTA', 'PUERTO LOPEZ', 'SALENTO',
       'FLORIDABLANCA', 'VENADILLO', 'FLORIDA', 'LA MESA', 'MANAURE',
       'CALARCA', 'CORDOBA', 'GALAN', 'LEBRIJA', 'SABANA DE TORRES',
       'BOGOTA, D.C.', 'MONTECRISTO', 'CHIVATA', 'CHOACHI',
       'PUERTO SALGAR', 'PEREIRA', 'MANIZALES', 'ARBELAEZ', 'FOMEQUE',
       'VILLAPINZON', 'VILLETA', 'COLON', 'SARDINATA', 'PUERTO CARREÑO',
       'PENSILVANIA', 'TAMARA', 'VALLEDUPAR', 'FONSECA', 'PUERTO ASIS',
       'DOSQUEBRADAS', 'CHAPARRAL', 'PUERTO BOYACA', 'CARMEN DEL DARIEN',
       'CONVENCION', 'TIQUISIO', 'PAZ DE ARIPORO', 'MALLAMA', 'HERVEO',
       'IBAGUE', 'PALOCABILDO', 'SAN ANTONIO', 'CALI', 'ZARZAL',
       'BARRANQUILLA', 'A

In [229]:
# Se verifica que los tipos de evento esten bien escritos
Basedatos["TIPO_DE_EVENTO"].unique()

array(['INCENDIO FORESTAL', 'INUNDACIÓN', 'DESLIZAMIENTO',
       'INCENDIO DE COBERTURA VEGETAL', 'VENDAVAL', 'MOVIMIENTO EN MASA',
       'INCENDIO ESTRUCTURAL', 'OTROS', 'ACCIDENTE', 'REMOCIÓN EN MASA',
       'TORMENTA ELECTRICA', 'ACCIDENTE DE TRÁNSITO', 'TEMPORAL',
       'AVENIDA TORRENCIAL', 'ACCIDENTE FLUVIAL', 'INUNDACION',
       'CRECIENTE SUBITA', 'CRECIENTE SÚBITA', 'EROSION', 'SEQUIA',
       'ACCIDENTE MINERO', 'ACCIDENTE TRANSPORTE TERRESTRE', 'INMERSION',
       'INCENDIO VEHICULAR', 'COLAPSO', 'AGLOMERACIÓN DE PÚBLICO',
       'MAR DE LEVA', 'ACCIDENTE TRANSPORTE AEREO', 'COLAPSO ESTRUCTURAL',
       'CONTAMINACIÓN', 'CASO FORTUITO', 'ACCIDENTE TRANSPORTE AÉREO',
       'GRANIZADA', 'EXPLOSION', 'EVENTO MASIVO', 'HELADA',
       'ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL', 'ACCIDENTE TECNOLOGICO',
       'SISMO', 'DERRAME', 'ACCIDENTE AEREO', 'ACCIDENTE POR CAIDA',
       'FUGA', 'MATPEL', 'FALLA ESTRUCTURAL', 'ACCIDENTE MARITIMO',
       'MAREJADA', 'ACCIDENTE AÉREO',

In [230]:
#Reemplazar las categorias mal escritas o con tildes para un mejor manejo y agrupar las categorias con pocos sucesos al tipo de evento mas conveniente
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['INCENDIO fORESTAL'],'INCENDIO FORESTAL')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['INCENDIO - OTROS','INCENDIO VEHICULAR','CONATO'],'INCENDIO')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['INUNDACIÓN','INUNDACIoN'],'INUNDACION')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['CONTAMINACIÓN'],'CONTAMINACION')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['REMOCIÓN EN MASA'],'REMOCION EN MASA')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['CRECIENTE SÚBITA'],'CRECIENTE SUBITA')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['ACCIDENTE DE TRÁNSITO'],'ACCIDENTE DE TRANSITO')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['IMERSION','INMERSIoN' ],'INMERSION')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['ACCIDENTE AEREO','ACCIDENTE TRANSPORTE AÉREO','ACCIDENTE AÉREO' ],'ACCIDENTE TRANSPORTE AEREO')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['ACCIDENTE FLUVIAL','ACCIDENTE MARITIMO','MAREA ALTA','NAUFRAGIO','MAREJADA' ],'ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['REPRESAMIENTO','DIAPIRISMO','AGLOMERACIÓN DE PÚBLICO','EVENTO MASIVO','ACCIDENTE POR CAIDA','EVENTO MASIVO','FALLA ESTRUCTURAL','ACCIDENTE ','EVENTO MAYOR','DERRAME' ],'ACCIDENTE')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['MAR DE LEVA','INTOXICACION MASIVA','ANTROPOGENICO NO INTENCIONAL','COVID-19','ACTIVACIÓN VOLCANICA','FUGA','EROSION COSTERA','MATPEL'],'OTROS')
Basedatos['TIPO_DE_EVENTO']=Basedatos['TIPO_DE_EVENTO'].replace(['CICLON TROPICAL: DEPRESION/TORMENTA/HURACAN'],'CICLON TROPICAL')

In [231]:
Basedatos['TIPO_DE_EVENTO'].value_counts()

INCENDIO DE COBERTURA VEGETAL              4113
INCENDIO FORESTAL                          1381
INUNDACION                                 1304
VENDAVAL                                   1042
MOVIMIENTO EN MASA                          845
INCENDIO ESTRUCTURAL                        717
DESLIZAMIENTO                               398
CRECIENTE SUBITA                            270
ACCIDENTE TRANSPORTE TERRESTRE              208
TEMPORAL                                    201
AVENIDA TORRENCIAL                          160
OTROS                                       112
REMOCION EN MASA                            106
COLAPSO ESTRUCTURAL                         104
ACCIDENTE MINERO                             96
ACCIDENTE DE TRANSITO                        76
EXPLOSION                                    67
SEQUIA                                       64
COLAPSO                                      51
ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL      48
ACCIDENTE TRANSPORTE AEREO              

<p><a name="s3"></a></p>

# **3. Tratamiento de datos nulos**

In [232]:
# Validar la cantidad de datos nulos
Basedatos.isnull().sum()

FECHA                   0
DEPARTAMENTO            0
MUNICIPIO               0
TIPO_DE_EVENTO          0
DIVIPOLA                0
FALLECIDOS              2
HERIDOS                 0
DESAPARECIDOS           0
AFECTADOS               0
FAMILIAS                0
VIVIENDAS_DESTRUIDAS    0
VIVIENDAS_AVERIADAS     0
LATITUD                 0
LONGITUD                0
dtype: int64

In [234]:
# Se observan los datos nulos de los fallecidos
Basedatos[Basedatos['FALLECIDOS'].isnull()] 

FECHA DEPARTAMENTO     MUNICIPIO                 TIPO_DE_EVENTO  \
1853 2020-05-19    ATLANTICO  BARRANQUILLA                     INUNDACION   
4301 2020-09-13    SANTANDER         GIRON  INCENDIO DE COBERTURA VEGETAL   

      DIVIPOLA  FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
1853    8001.0         NaN        0              0        764       191   
4301   68307.0         NaN        0              0          0         0   

      VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS    LATITUD   LONGITUD  
1853                     0                   71  10.981521 -74.827715  
4301                     0                    0   6.976734 -73.205267

In [235]:
# Nuevamente al ser solo 2 datos nulos se buscó el numero de fallecidos en cada evento manualmente, encontrando que no hubo fallecidos por lo cual se ponen como cero
Basedatos.iloc[1853,5] = "0"
Basedatos.iloc[4301,5] = "0"

In [236]:
# Se confirma que ya no hay nulos
Basedatos[Basedatos['FALLECIDOS'].isnull()] 

Empty DataFrame
Columns: [FECHA, DEPARTAMENTO, MUNICIPIO, TIPO_DE_EVENTO, DIVIPOLA, FALLECIDOS, HERIDOS, DESAPARECIDOS, AFECTADOS, FAMILIAS, VIVIENDAS_DESTRUIDAS, VIVIENDAS_AVERIADAS, LATITUD, LONGITUD]
Index: []

In [237]:
# Validar la cantidad de datos nulos
Basedatos.isnull().sum()

FECHA                   0
DEPARTAMENTO            0
MUNICIPIO               0
TIPO_DE_EVENTO          0
DIVIPOLA                0
FALLECIDOS              0
HERIDOS                 0
DESAPARECIDOS           0
AFECTADOS               0
FAMILIAS                0
VIVIENDAS_DESTRUIDAS    0
VIVIENDAS_AVERIADAS     0
LATITUD                 0
LONGITUD                0
dtype: int64

###**Cambiar el tipo de dato de las columnas por el correcto**

In [240]:
# Se identifica que divipola y fallecidos están en el formato incorrecto
Basedatos.dtypes

FECHA                   datetime64[ns]
DEPARTAMENTO                    object
MUNICIPIO                       object
TIPO_DE_EVENTO                  object
DIVIPOLA                       float64
FALLECIDOS                      object
HERIDOS                          int64
DESAPARECIDOS                    int64
AFECTADOS                        int64
FAMILIAS                         int64
VIVIENDAS_DESTRUIDAS             int64
VIVIENDAS_AVERIADAS              int64
LATITUD                        float64
LONGITUD                       float64
dtype: object

In [241]:
# Se cambian al formato correcto (entero)
Basedatos['FALLECIDOS'] = Basedatos['FALLECIDOS'].apply(np.int64)
Basedatos['DIVIPOLA'] = Basedatos['DIVIPOLA'].apply(np.int64)

In [242]:
Basedatos.dtypes

FECHA                   datetime64[ns]
DEPARTAMENTO                    object
MUNICIPIO                       object
TIPO_DE_EVENTO                  object
DIVIPOLA                         int64
FALLECIDOS                       int64
HERIDOS                          int64
DESAPARECIDOS                    int64
AFECTADOS                        int64
FAMILIAS                         int64
VIVIENDAS_DESTRUIDAS             int64
VIVIENDAS_AVERIADAS              int64
LATITUD                        float64
LONGITUD                       float64
dtype: object

###**Verificación valores coherentes**

In [244]:
# Se verifica que no existan fechas incoherentes
print(Basedatos['FECHA'].min())
print(Basedatos['FECHA'].max())

2008-12-31 00:00:00
2020-12-31 00:00:00


In [245]:
#se observa que la menor fecha es del año 2008 y las base de datos es del 2018,2019 y 2020
Basedatos[Basedatos['FECHA']== '2008-12-31 00:00:00']

FECHA DEPARTAMENTO  MUNICIPIO     TIPO_DE_EVENTO  DIVIPOLA  \
9578 2008-12-31       NARIÑO  SAN PABLO  INCENDIO FORESTAL     52693   

      FALLECIDOS  HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  \
9578           0        0              0          0         0   

      VIVIENDAS_DESTRUIDAS  VIVIENDAS_AVERIADAS   LATITUD  LONGITUD  
9578                     0                    0  1.681577 -76.97528

In [246]:
# Se elimina ya que no tiene ningun registro que sea relevante
Basedatos= Basedatos.drop([9578])

In [248]:
#Se confirma que la fecha minima cambio al primero de enero del 2018
Basedatos['FECHA'].min()

Timestamp('2018-01-01 00:00:00')

In [249]:
# Tambien se revisara el dato alto en AFECTADOS
Basedatos['AFECTADOS'].max()

150000

In [250]:
# Se buscó informacion sobre inundaciones en Cartagena en noviembre del 2020 donde si hubieron bastantes afectados por el huracan Iota, pero posiblemente
# en este día 2020-11-13, se registraron todos los damnificados absolutos. Es por esto que se decide eliminar este dato ya que puede sesgar los resultados
# del estudio.
# Fuente: https://www.lafm.com.co/colombia/tormenta-iota-deja-mas-de-150000-damnificados-en-cartagena
Basedatos[Basedatos['AFECTADOS'] == 150000] 

FECHA DEPARTAMENTO  MUNICIPIO TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  \
2401 2020-11-13      BOLIVAR  CARTAGENA     INUNDACION     13001           3   

      HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
2401        0              0     150000     30000                    14   

      VIVIENDAS_AVERIADAS    LATITUD   LONGITUD  
2401                    3  10.463434 -75.458899

In [251]:
 # Se borra la fila
Basedatos = Basedatos.drop([2401])

In [252]:
Basedatos['AFECTADOS'].max()

25408

In [253]:
# Si hay datos duplicados (se eliminan 672 datos)
print(Basedatos.shape)
print(Basedatos.drop_duplicates().shape)

(11690, 14)
(11018, 14)


<p><a name="s3"></a></p>

# **4. Generación de bodegas de datos**

- En esta parte se crea la serie de tiempo de las fechas y un DataFrame de la base de datos completa con la serie de las fechas en orden (Seriefecha y base_datos).
- Por otro lado, se filtra la bodega de datos con los tipos de eventos que tuvieron causas naturales, para un mejor analisis ya que el estudio se centra en el medio ambiente.
- También se crearon bodegas de datos en el interior del ánalisis de cada pregunta.

In [254]:
# Se identifican los eventos
Basedatos['TIPO_DE_EVENTO'].unique()

array(['INCENDIO FORESTAL', 'INUNDACION', 'DESLIZAMIENTO',
       'INCENDIO DE COBERTURA VEGETAL', 'VENDAVAL', 'MOVIMIENTO EN MASA',
       'INCENDIO ESTRUCTURAL', 'OTROS', 'ACCIDENTE', 'REMOCION EN MASA',
       'TORMENTA ELECTRICA', 'ACCIDENTE DE TRANSITO', 'TEMPORAL',
       'AVENIDA TORRENCIAL', 'ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL',
       'CRECIENTE SUBITA', 'EROSION', 'SEQUIA', 'ACCIDENTE MINERO',
       'ACCIDENTE TRANSPORTE TERRESTRE', 'INMERSION', 'INCENDIO',
       'COLAPSO', 'ACCIDENTE TRANSPORTE AEREO', 'COLAPSO ESTRUCTURAL',
       'CONTAMINACION', 'CASO FORTUITO', 'GRANIZADA', 'EXPLOSION',
       'HELADA', 'ACCIDENTE TECNOLOGICO', 'SISMO', 'CICLON TROPICAL'],
      dtype=object)

**Creación de la bodega de datos**

In [256]:
#Solo se toman los eventos que pudieron haber tenido causas naturales o que estén más relacionados con el tema principal de estudio que es el medio ambiente
#por lo tanto de Basedatos se decide eliminar ACCIDENTE DE TRANSITO, ACCIDENTE TRANSPORTE AEREO y ACCIDENTE TECNOLOGICO.

datos = Basedatos[Basedatos['TIPO_DE_EVENTO'].isin(['INCENDIO FORESTAL', 'DESLIZAMIENTO',
       'ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL', 'ACCIDENTE',
       'CRECIENTE SUBITA', 'VENDAVAL', 'INCENDIO ESTRUCTURAL',
       'INUNDACION', 'REMOCION EN MASA', 'OTROS', 'ACCIDENTE MINERO',
       'GRANIZADA', 'EXPLOSION', 'COLAPSO', 'SEQUIA', 'AVENIDA TORRENCIAL',
       'CONTAMINACION', 'TORMENTA ELECTRICA', 'INCENDIO', 'EROSION',
       'SISMO', 'INMERSION', 'INCENDIO DE COBERTURA VEGETAL',
       'COLAPSO ESTRUCTURAL', 'MOVIMIENTO EN MASA',
       'ACCIDENTE TRANSPORTE TERRESTRE',
       'TEMPORAL', 'CASO FORTUITO', 'HELADA','CICLON TROPICAL'])]


datos.head()

FECHA DEPARTAMENTO MUNICIPIO     TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  \
0 2018-01-01       BOYACA  GUATEQUE  INCENDIO FORESTAL     15322           0   
1 2018-01-28       BOYACA  GUATEQUE  INCENDIO FORESTAL     15322           0   
2 2018-02-09       BOYACA  GUATEQUE  INCENDIO FORESTAL     15322           0   
3 2018-02-09       BOYACA  GUATEQUE  INCENDIO FORESTAL     15322           0   
4 2018-02-26       BOYACA  GUATEQUE  INCENDIO FORESTAL     15322           0   

   HERIDOS  DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
0        0              0          0         0                     0   
1        0              0          0         0                     0   
2        0              0          0         0                     0   
3        0              0          0         0                     0   
4        0              0          0         0                     0   

   VIVIENDAS_AVERIADAS   LATITUD   LONGITUD  
0                    0  5.014508 -73.488124  
1                    0  5.014508 -73.488124  
2                    0  5.014508 -73.488124  
3                    0  5.014508 -73.488124  
4                    0  5.014508 -73.488124

In [258]:
# Se creo un nuevo dataframe con las fechas en orden para asegurarnos de tener una serie de tiempo
Seriefecha= pd.DataFrame(pd.date_range(start=datos['FECHA'].min(), end=datos['FECHA'].max())).rename(columns ={0:'FECHA'})
Seriefecha

FECHA
0    2018-01-01
1    2018-01-02
2    2018-01-03
3    2018-01-04
4    2018-01-05
...         ...
1091 2020-12-27
1092 2020-12-28
1093 2020-12-29
1094 2020-12-30
1095 2020-12-31

[1096 rows x 1 columns]

In [259]:
#Se unio la base de datos completa con las fechas en orden
base_datos = pd.merge(Seriefecha, datos, on ='FECHA', how ='inner')

In [261]:
base_datos.head()

FECHA  DEPARTAMENTO   MUNICIPIO  \
0 2018-01-01        BOYACA    GUATEQUE   
1 2018-01-01  CUNDINAMARCA      GUASCA   
2 2018-01-01       QUINDIO  MONTENEGRO   
3 2018-01-02     ANTIOQUIA       ANORI   
4 2018-01-02         CHOCO       LLORO   

                            TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  HERIDOS  \
0                        INCENDIO FORESTAL     15322           0        0   
1                            DESLIZAMIENTO     25322           0        0   
2                            DESLIZAMIENTO     63470           0        1   
3                            DESLIZAMIENTO      5040           0        0   
4  ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL     27413           2        0   

   DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
0              0          0         0                     0   
1              0          0         0                     0   
2              0          1         0                     0   
3              0         15         3                     1   
4              0          2         0                     0   

   VIVIENDAS_AVERIADAS   LATITUD   LONGITUD  
0                    0  5.014508 -73.488124  
1                    0  4.801277 -73.846435  
2                    0  4.521891 -75.818066  
3                    2  7.193111 -75.108490  
4                    0  5.585101 -76.385444

<p><a name="s3"></a></p>

# **5. Transformaciones: Aplicación de funciones para la creación de nuevas variables**

In [262]:
# Se confirma que los datos a sumar para crear una nueva columna que contenga el total de personas involucrados en dichos eventos sean compatibles
datos.dtypes 

FECHA                   datetime64[ns]
DEPARTAMENTO                    object
MUNICIPIO                       object
TIPO_DE_EVENTO                  object
DIVIPOLA                         int64
FALLECIDOS                       int64
HERIDOS                          int64
DESAPARECIDOS                    int64
AFECTADOS                        int64
FAMILIAS                         int64
VIVIENDAS_DESTRUIDAS             int64
VIVIENDAS_AVERIADAS              int64
LATITUD                        float64
LONGITUD                       float64
dtype: object

In [263]:
# Se crea una columna con los resultados de las columnas heridos, fallecidos y desaparecidos, llamada personas involucradas
# y que de como resultado en cada fila (SI ó NO).
# de esta manera se observa si en el evento ocurrido, existieron personas involucradas 

tranformacion = [ 
    ((base_datos.loc[:,'DESAPARECIDOS']  + base_datos.loc[:,'HERIDOS'] + base_datos.loc[:,'FALLECIDOS']) > 0 ),
    ((base_datos.loc[:,'DESAPARECIDOS']  + base_datos.loc[:,'HERIDOS'] + base_datos.loc[:,'FALLECIDOS']) <= 0)]
    
values1 = ['SI','NO']
base_datos['PERSONAS_INVOLUCRADAS']= np.select(tranformacion, values1)
base_datos

FECHA  DEPARTAMENTO            MUNICIPIO  \
0     2018-01-01        BOYACA             GUATEQUE   
1     2018-01-01  CUNDINAMARCA               GUASCA   
2     2018-01-01       QUINDIO           MONTENEGRO   
3     2018-01-02     ANTIOQUIA                ANORI   
4     2018-01-02         CHOCO                LLORO   
...          ...           ...                  ...   
11554 2020-12-30         CAUCA               BALBOA   
11555 2020-12-31     MAGDALENA        NUEVA GRANADA   
11556 2020-12-31     RISARALDA  SANTA ROSA DE CABAL   
11557 2020-12-31         CAUCA              POPAYAN   
11558 2020-12-31         CAUCA              POPAYAN   

                                TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  HERIDOS  \
0                            INCENDIO FORESTAL     15322           0        0   
1                                DESLIZAMIENTO     25322           0        0   
2                                DESLIZAMIENTO     63470           0        1   
3                                DESLIZAMIENTO      5040           0        0   
4      ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL     27413           2        0   
...                                        ...       ...         ...      ...   
11554                       MOVIMIENTO EN MASA     19075           0        0   
11555                     INCENDIO ESTRUCTURAL     47460           0        0   
11556           ACCIDENTE TRANSPORTE TERRESTRE     66682           0        1   
11557                               INUNDACION     19001           0        0   
11558                               INUNDACION     19001           0        0   

       DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
0                  0          0         0                     0   
1                  0          0         0                     0   
2                  0          1         0                     0   
3                  0         15         3                     1   
4                  0          2         0                     0   
...              ...        ...       ...                   ...   
11554              0         15         3                     0   
11555              0          0         0                     0   
11556              0          1         0                     0   
11557              0          0         0                     0   
11558              0         20         5                     0   

       VIVIENDAS_AVERIADAS   LATITUD   LONGITUD PERSONAS_INVOLUCRADAS  
0                        0  5.014508 -73.488124                    NO  
1                        0  4.801277 -73.846435                    NO  
2                        0  4.521891 -75.818066                    SI  
3                        2  7.193111 -75.108490                    NO  
4                        0  5.585101 -76.385444                    SI  
...                    ...       ...        ...                   ...  
11554                    3  2.084733 -77.208813                    NO  
11555                    0  9.735998 -74.328402                    NO  
11556                    0  4.833790 -75.549502                    SI  
11557                    0  2.471704 -76.591944                    NO  
11558                    5  2.471704 -76.591944                    NO  

[11559 rows x 15 columns]

In [264]:
# Teniendo en cuenta el numero de personas afectadas, se crea una columna que se llama (NIVEL_DEL_INCIDENTE_NATURAL) y 
# que da como resultado de:
# 0 a 100 afectados (LEVE)
# 101 a 1000 afectados (IMPORTANTE)
# 1001 a 10000 afectados( GRAVE) 
# superior (URGENTE)

tranformacion2 = [ 
    ((base_datos.loc[:,'AFECTADOS']  ) <= 100 ),
    (base_datos.loc[:,'AFECTADOS']   > 101) & (base_datos.loc[:,'AFECTADOS']   <= 1000),
    (base_datos.loc[:,'AFECTADOS']   > 1001) & (base_datos.loc[:,'AFECTADOS']  <= 10000),
    (base_datos.loc[:,'AFECTADOS']   > 10001 )
    ]
    
values12 = ['LEVE','IMPORTANTE','GRAVE','URGENTE']
base_datos['NIVEL_DEL_INCIDENTE_NATURAL']= np.select(tranformacion2, values12)
base_datos


FECHA  DEPARTAMENTO            MUNICIPIO  \
0     2018-01-01        BOYACA             GUATEQUE   
1     2018-01-01  CUNDINAMARCA               GUASCA   
2     2018-01-01       QUINDIO           MONTENEGRO   
3     2018-01-02     ANTIOQUIA                ANORI   
4     2018-01-02         CHOCO                LLORO   
...          ...           ...                  ...   
11554 2020-12-30         CAUCA               BALBOA   
11555 2020-12-31     MAGDALENA        NUEVA GRANADA   
11556 2020-12-31     RISARALDA  SANTA ROSA DE CABAL   
11557 2020-12-31         CAUCA              POPAYAN   
11558 2020-12-31         CAUCA              POPAYAN   

                                TIPO_DE_EVENTO  DIVIPOLA  FALLECIDOS  HERIDOS  \
0                            INCENDIO FORESTAL     15322           0        0   
1                                DESLIZAMIENTO     25322           0        0   
2                                DESLIZAMIENTO     63470           0        1   
3                                DESLIZAMIENTO      5040           0        0   
4      ACCIDENTE TRANSPORTE MARITIMO O FLUVIAL     27413           2        0   
...                                        ...       ...         ...      ...   
11554                       MOVIMIENTO EN MASA     19075           0        0   
11555                     INCENDIO ESTRUCTURAL     47460           0        0   
11556           ACCIDENTE TRANSPORTE TERRESTRE     66682           0        1   
11557                               INUNDACION     19001           0        0   
11558                               INUNDACION     19001           0        0   

       DESAPARECIDOS  AFECTADOS  FAMILIAS  VIVIENDAS_DESTRUIDAS  \
0                  0          0         0                     0   
1                  0          0         0                     0   
2                  0          1         0                     0   
3                  0         15         3                     1   
4                  0          2         0                     0   
...              ...        ...       ...                   ...   
11554              0         15         3                     0   
11555              0          0         0                     0   
11556              0          1         0                     0   
11557              0          0         0                     0   
11558              0         20         5                     0   

       VIVIENDAS_AVERIADAS   LATITUD   LONGITUD PERSONAS_INVOLUCRADAS  \
0                        0  5.014508 -73.488124                    NO   
1                        0  4.801277 -73.846435                    NO   
2                        0  4.521891 -75.818066                    SI   
3                        2  7.193111 -75.108490                    NO   
4                        0  5.585101 -76.385444                    SI   
...                    ...       ...        ...                   ...   
11554                    3  2.084733 -77.208813                    NO   
11555                    0  9.735998 -74.328402                    NO   
11556                    0  4.833790 -75.549502                    SI   
11557                    0  2.471704 -76.591944                    NO   
11558                    5  2.471704 -76.591944                    NO   

      NIVEL_DEL_INCIDENTE_NATURAL  
0                            LEVE  
1                            LEVE  
2                            LEVE  
3                            LEVE  
4                            LEVE  
...                           ...  
11554                        LEVE  
11555                        LEVE  
11556                        LEVE  
11557                        LEVE  
11558                        LEVE  

[11559 rows x 16 columns]

<p><a name="s3"></a></p>

# **6. Preguntas**

###**- Pregunta 1: Muestre un gráfico con el número de personas afectadas por departamentos desde el 2018 al 2020**

In [271]:
#Agrupar por departamentos el número de personas afectadas
df1 = base_datos.groupby(['DEPARTAMENTO'])[['AFECTADOS']].sum().reset_index().sort_values('AFECTADOS', ascending = False)

# crear gráfica
fig = px.bar(df1, x='DEPARTAMENTO',y='AFECTADOS', barmode= 'group', color_discrete_sequence=px.colors.qualitative.Set3,text_auto='.2s')

# agregar detalles a la gráfica
fig.update_layout(
        xaxis_title = 'Departamento',
        yaxis_title = 'Cantidad de personas afectadas',
        template = 'simple_white',
        title_x = 0.5)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)

fig.show()

**Interpretación:** Teniendo en cuenta los 32 departamentos que tiene Colombia, los que presentan más personas afectadas por los desastres naturales son Chocó, La Guajira y Bolivar, cada uno con 252.993, 115.918 y 99.844 personas afectadas respectivamente. Lo que indica que estos tres departamentos son los más propensos a presenciar este tipo de desastres por diferentes variables como su ubicación geográfica o su situación política ya que son zonas marginadas. Además se destaca que Amazonas es el departamento en ocupar el último lugar, probablemente se deba a la falta de información en los registros.

###**- Pregunta 2: De los departamentos más afectados, ¿Cuántas son las personas afectadas según el Nivel del Incidente Natural?**

In [290]:
# Crear base de datos filtrando por los 3 departamentos con más afectados (Chocó, La Guajira, Bolivar)
df0=base_datos[['DEPARTAMENTO','NIVEL_DEL_INCIDENTE_NATURAL','AFECTADOS']]
df0=df0[df0['DEPARTAMENTO'].isin(['CHOCO','LA GUAJIRA','BOLIVAR'])]

#Agrupar por departamento y nivel del incidente
df0=df0.groupby(['DEPARTAMENTO','NIVEL_DEL_INCIDENTE_NATURAL'])[['AFECTADOS']].sum().reset_index().sort_values('AFECTADOS', ascending = False).rename(columns={'NIVEL_DEL_INCIDENTE_NATURAL':'Nivel del Incidente'})
df0

DEPARTAMENTO Nivel del Incidente  AFECTADOS
4         CHOCO               GRAVE     205202
0       BOLIVAR               GRAVE      60446
11   LA GUAJIRA             URGENTE      57758
8    LA GUAJIRA               GRAVE      52208
7         CHOCO             URGENTE      25000
5         CHOCO          IMPORTANTE      21382
3       BOLIVAR             URGENTE      20690
1       BOLIVAR          IMPORTANTE      17443
9    LA GUAJIRA          IMPORTANTE       5114
6         CHOCO                LEVE       1409
2       BOLIVAR                LEVE       1265
10   LA GUAJIRA                LEVE        838

In [293]:
# Grafica
df = px.data.tips()
fig = px.bar(df0, x='Nivel del Incidente', y='AFECTADOS', barmode="group", facet_col='DEPARTAMENTO', color_discrete_sequence=px.colors.qualitative.Pastel1,
             text_auto='.2s')

# agregar detalles a la gráfica
fig.update_layout(
        yaxis_title = 'Cantidad de personas afectadas',
        template = 'simple_white',
        title_x = 0.5)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)

fig.show()


**Interpretación:** De los departamentos analizados anteriormente (Chocó, La Guajira y Bolivar), se encuentra que los 3 tienen alta presencia de incidentes de categoria grave y urgente, ya que la cantidad de personas afectadas es mucha a comparación de las otras categorias como importante y leve. Lo que confirma la hipotesis anterior de que estos tres departamentos son más suceptibles a sufrir graves accidentes.

###**- Pregunta 3: ¿Cuántos desaparecidos, fallecidos y heridos hubo por año en Colombia?**

In [341]:
# Crear la base de datos
df5 = base_datos[['FECHA','FALLECIDOS','DESAPARECIDOS','HERIDOS']]	
	
df5=df5.resample('A',on ='FECHA').sum().reset_index() # Reagrupa por mes y suma los valores que comprende este periodo
df5['FECHA']=df5['FECHA'].dt.year

In [342]:
df5

FECHA  FALLECIDOS  DESAPARECIDOS  HERIDOS
0   2018         310             43      819
1   2019         368             27     1665
2   2020         355             30      880

In [345]:
# crear gráfica
fig = px.bar(df5, x='FECHA', y=['HERIDOS','FALLECIDOS','DESAPARECIDOS'], barmode ='group',
             color_discrete_sequence=px.colors.qualitative.Set2,
             text_auto='.2s',
             title ='<b>Personas desaparecidas, heridas y fallecidas entre el 2018 y 2020<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Año',
    yaxis_title = 'Número de personas',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b>Estado<b>')
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)

fig.show()

**Interpretación:** Al comparar el número de personas heridas, fallecidas y desaparecidas por año, se evidencia que todos los años (2018, 2019 y 2020) tienen el mismo comportamiento; existe una mayor cantidas de personas heridas, luego siguen los fallecidos y por último se encuentran los desaparecidos. En el año 2019 se registra el mayor número de personas afectadas con 1665 heridos, por lo cual se podría afirmar que en este año sucedieron eventos que fueron más tragicos para las personas que en los demás años ya que el número de personas fallecidas también fue mayor.

###**- Pregunta 4: ¿Cuál es el tipo evento que tiene mas afectados y en qué porcentaje?**

In [300]:
# importar paquete
import plotly.graph_objects as go 

# crear base
df0 = base_datos.groupby(['TIPO_DE_EVENTO'])[['HERIDOS']].sum().sort_values('HERIDOS', ascending = False).rename(columns={'HERIDOS':'counts'})
df0['ratio'] = df0.apply(lambda x: x.cumsum()/df0['counts'].sum())

# definir figura
fig = go.Figure([go.Bar(x=df0.index, y=df0['counts'], yaxis='y1', name='sessions id'),
                 go.Scatter(x=df0.index, y=df0['ratio'], yaxis='y2', name='Heridos', hovertemplate='%{y:.1%}', marker={'color': '#000000'})])

# agregar detalles
fig.update_layout(template='plotly_white', showlegend=False, hovermode='x', bargap=.3,
                  title={'text': '<b> Heridos por tipo de evento<b>', 'x': .5}, 
                  yaxis={'title': 'Heridos'},
                  yaxis2={'rangemode': "tozero", 'overlaying': 'y', 'position': 1, 'side': 'right', 'title': 'ratio', 'tickvals': np.arange(0, 1.1, .2), 'tickmode': 'array', 'ticktext': [str(i) + '%' for i in range(0, 101, 20)]})

fig.show()

**Interpretación:** Según el gráfico de pareto, los eventos de accidente transporte terrestre, incendio estructural y explosión son los principales responsables del gran número de personas heridas que se analizó en la gráfica anterior. Estos 3 eventos tienen un 60,8% de participación en la afectación a las personas. Además se resalta que en estos eventos es probable que exista la participación del hombre aunque sea de manera intencional, ya sea por una imprudencia al conducir, un incendió que haya tenido un inicio en una vivienda o artefactos mal manipulados que al final tienen consecuencias en el medio ambiente.

###**- Pregunta 5: ¿Cuál es la proporción de personas afectadas, fallecidas y heridas por los eventos incendio de cobertura vegetal, incendio forestal, incendio estructural e incendio?**

**Personas afectadas**

In [302]:
# Crear la base de datos
base2 = base_datos[base_datos['TIPO_DE_EVENTO'].isin(['INCENDIO DE COBERTURA VEGETAL','INCENDIO FORESTAL','INCENDIO ESTRUCTURAL','INCENDIO'])]

# crear gráfica:
fig = px.pie(base2 , values = 'AFECTADOS', names = 'TIPO_DE_EVENTO',
             color_discrete_sequence=px.colors.qualitative.Pastel,
             title = '<b>% Personas afectadas por cada tipo de incendio <b>')

# agregar detalles a la gráfica:
fig.update_layout(
    template = 'simple_white',
    legend_title = '<b>Tipo de incendio<b>',
    title_x = 0.5)

fig.show()

**Interpretación:** La proporción más grande de personas afectadas debido a los incendios fue por el tipo de incendio estructural, este tipo de incendio como se mencionó anteriormente, puede tener causas como accidentes domésticos, fallas eléctricas o fugas de gases, en las cuales el hombre pudo haber tenido una intervención de manera no intencional.
Luego están los incendios forestales y de cobertura vegetal, los cuales si tienen causas más naturales como la caída de rayos o erupciones volcánicas.


**Personas fallecidas**

In [303]:
# Crear la base de datos
base2 = base_datos[base_datos['TIPO_DE_EVENTO'].isin(['INCENDIO DE COBERTURA VEGETAL','INCENDIO FORESTAL','INCENDIO ESTRUCTURAL','INCENDIO'])]

# crear gráfica:
fig = px.pie(base2 , values = 'FALLECIDOS', names = 'TIPO_DE_EVENTO',
             color_discrete_sequence=px.colors.qualitative.Pastel1,
             title = '<b>% Personas fallecidas por cada tipo de incendio <b>')

# agregar detalles a la gráfica:
fig.update_layout(
    template = 'simple_white',
    legend_title = '<b>Tipo de incendio<b>',
    title_x = 0.5)

fig.show()

**Interpretación:** La mayor proporción de personas fallecidas también es debido a los incendios estructurales con una presencia del 89,1%, mientras que los incendios de cobertura vegetal, forestal e incendio (otro tipo) tuvieron una participación muy baja.

**Personas heridas**

In [304]:
# Crear la base de datos
base2 = base_datos[base_datos['TIPO_DE_EVENTO'].isin(['INCENDIO DE COBERTURA VEGETAL','INCENDIO FORESTAL','INCENDIO ESTRUCTURAL','INCENDIO'])]

# crear gráfica:
fig = px.pie(base2 , values = 'HERIDOS', names = 'TIPO_DE_EVENTO',
             color_discrete_sequence=px.colors.qualitative.Set3,
             title = '<b>% Personas heridas por cada tipo de incendio <b>')

# agregar detalles a la gráfica:
fig.update_layout(
    template = 'simple_white',
    legend_title = '<b>Tipo de incendio<b>',
    title_x = 0.5)

fig.show()

**Interpretación:** Al igual que en los demás análisis, el incendio estructural fue el tipo de incendio que causó más personas heridas y el incendio de cobertura vegetal tuvo una presencia considerable con un 9,21%. El incendio forestal no fue muy significativo y el otro tipo de incendio no tuvo registros.

###**- Pregunta 6: ¿Cuál es la cantidad de veces en que sucedió cada evento durante todo el periodo de tiempo (Top 10)?**

In [312]:
#Crear base de datos agrupando por evento y contando las fechas en que sucedió cada uno
cuenta=base_datos.groupby(['TIPO_DE_EVENTO'])[['FECHA']].count().rename(columns={'FECHA':'CANTIDAD'}).sort_values('CANTIDAD',ascending=False).reset_index().head(10)
cuenta

TIPO_DE_EVENTO  CANTIDAD
0   INCENDIO DE COBERTURA VEGETAL      4113
1               INCENDIO FORESTAL      1380
2                      INUNDACION      1303
3                        VENDAVAL      1042
4              MOVIMIENTO EN MASA       845
5            INCENDIO ESTRUCTURAL       717
6                   DESLIZAMIENTO       398
7                CRECIENTE SUBITA       270
8  ACCIDENTE TRANSPORTE TERRESTRE       208
9                        TEMPORAL       201

In [314]:
# crear gráfica
fig = px.bar(cuenta, x='TIPO_DE_EVENTO', y='CANTIDAD', 
             color_discrete_sequence=px.colors.qualitative.Bold,
             title ='<b>Top 10 cantidad de veces que sucedió cada evento<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Tipo de evento',
    yaxis_title = 'Cantidad',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b>Tipo de edificio<b>')

fig.show()

**Interpretación:** En este top 10 de eventos que más se presentaron en los años 2018, 2019 y 2020, el incendio de cobertura vegetal fue el principal ya que ocurrió 4113 veces en estos tres años y en segundo lugar se encuentra el incendio forestal con un total de 1380 veces. Ambos eventos pueden haber tenido causas iguales y las consecuencias en la mayoría de los casos es la destrucción del ecosistema Colombiano y el deterioro del medio ambiente. En tercer lugar están las inundaciones que también ocurren muy frecuentemente en Colombia, ya que en estos últimos años han ocurrido varios huracanes y tormentas afectando a los departamentos más cercanos al mar.

###**- Pregunta 7: ¿Cuál es el mes en el que ocurren más viviendas destruidas y averiadas?**

**Viviendas destruidas**

In [327]:
# Crear base de datos seleccionando solo el mes y agrupando por el mismo
casas=base_datos.loc[:,['FECHA','VIVIENDAS_DESTRUIDAS']]
casas['FECHA']=casas['FECHA'].dt.month
casas=casas.groupby(['FECHA'])[['VIVIENDAS_DESTRUIDAS']].sum().reset_index()
casas

FECHA  VIVIENDAS_DESTRUIDAS
0       1                   153
1       2                   245
2       3                   268
3       4                   248
4       5                   275
5       6                   199
6       7                   245
7       8                   211
8       9                   222
9      10                   388
10     11                  1963
11     12                   240

In [328]:
# Crear grafica
fig = px.bar(casas, x = 'FECHA', y='VIVIENDAS_DESTRUIDAS',title = '<b>Viviendas destruidas<b>',barmode= 'group', text_auto='.2s',
                color_discrete_sequence=px.colors.qualitative.Pastel)

fig.update_layout(
    xaxis_title = 'Fecha',
    yaxis_title = 'Número viviendas',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

**Interpretación:** Según el gráfico, el número de viviendas destruidas en cada mes tiene una media cercana a 250 excluyendo el mes 11 (noviembre), ya que en este mes fue donde más viviendas destruidas ocurrieron, para un total de 1963 viviendas y está muy por encima de los demás meses.

**Viviendas averiadas**

In [322]:
# Crear base de datos seleccionando solo el mes y agrupando por el mismo
casas=base_datos.loc[:,['FECHA','VIVIENDAS_AVERIADAS']]
casas['FECHA']=casas['FECHA'].dt.month
casas=casas.groupby(['FECHA'])[['VIVIENDAS_AVERIADAS']].sum().reset_index()
casas

FECHA  VIVIENDAS_AVERIADAS
0       1                 2574
1       2                 4301
2       3                 5662
3       4                 9569
4       5                14762
5       6                16070
6       7                 8382
7       8                 9140
8       9                12461
9      10                23757
10     11                29269
11     12                 4552

In [324]:
# Crear grafica
fig = px.bar(casas, x = 'FECHA', y='VIVIENDAS_AVERIADAS',title = '<b>Viviendas averiadas<b>',barmode= 'group', text_auto='.2s',
                color_discrete_sequence=px.colors.qualitative.Pastel1)

fig.update_layout(
    xaxis_title = 'Fecha',
    yaxis_title = 'Número viviendas',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

**Interpretación:** En comparación con las viviendas destruidas, las viviendas afectadas se presentaron en mayor cantidad. Además se nota el mismo patrón cada 6 meses, en donde desde enero a junio el número de viviendas afectadas va creciendo y de julio a noviembre ocurre lo mismo. Otro punto que se identificó a través del gráfico, fue que al igual que las viviendas destruidas, en el mes 11 (Noviembre) ocurrió el mayor número de viviendas afectadas, por lo cual, se puede concluir que en este mes se debe tener un mayor control y apoyo hacia las comunidades que más se ven afectadas ya que los eventos naturales que ocurren especialmente en esta epoca, ocasionan muchos daños a los colombianos.

###**- Pregunta 8: ¿Cuál es la relación de las viviendas averiadas y destruidas? Muestre los afectados según el departamento**

In [347]:
# Se agrupa por evento y departamento, luegos se suman las columnas de viviendas destruidas, averiadas y afectados
df0=base_datos.groupby(['TIPO_DE_EVENTO','DEPARTAMENTO']).agg(
    destruidas = ('VIVIENDAS_DESTRUIDAS','sum'),
    averiadas = ('VIVIENDAS_AVERIADAS','sum'),
    afectados = ('AFECTADOS','sum')
    ).reset_index().sort_values('afectados', ascending = False).head(10)
df0 = df0[df0['TIPO_DE_EVENTO']=='INUNDACION'] # Filtrar DataFrame

# definir gráfica
fig = px.scatter(df0, x = 'destruidas', y ='averiadas', color = 'DEPARTAMENTO', size = 'afectados')

# añadir atributos
fig.update_layout(
    xaxis_title = '<b>Viviendas destruidas<b>',
    yaxis_title = '<b>Viviendas averiadas<b>',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b> Departamento <b>')

fig.show()

**Interpretación:**Ya que la mayoria de viviendas averiadas y destruidas ocurren por las inundaciones en Colombia, a través del gráfico de dispersión, se logra analizar que las inundaciones ocasionan más viviendas averiadas que destruidas y que además, los departamentos que ya se han mencionado anteriormente como Chocó, La Guajira y Bolivar son los principales afectados por su ubicación y la calidad de vida de las personas, dejando muchos afectados.

###**- Pregunta 9: Top 7 de los departamentos con más personas involucradas**

In [354]:
# Se crea la base de datos
df0=base_datos.groupby(['PERSONAS_INVOLUCRADAS','DEPARTAMENTO']).agg(
    heridos = ('HERIDOS','sum'),
    desaparecidos = ('DESAPARECIDOS','sum'),
    fallecidos = ('FALLECIDOS','sum'),
    eventos = ('TIPO_DE_EVENTO','count')
    ).reset_index()
df0


PERSONAS_INVOLUCRADAS     DEPARTAMENTO  heridos  desaparecidos  fallecidos  \
0                     NO         AMAZONAS        0              0           0   
1                     NO        ANTIOQUIA        0              0           0   
2                     NO           ARAUCA        0              0           0   
3                     NO        ATLANTICO        0              0           0   
4                     NO          BOLIVAR        0              0           0   
..                   ...              ...      ...            ...         ...   
59                    SI            SUCRE        2              0           2   
60                    SI           TOLIMA      190             13          52   
61                    SI  VALLE DEL CAUCA      135              5          33   
62                    SI           VAUPES        0              0           8   
63                    SI          VICHADA        0              0           2   

    eventos  
0         6  
1       746  
2       137  
3       145  
4       215  
..      ...  
59        3  
60       75  
61       41  
62        6  
63        2  

[64 rows x 6 columns]

In [355]:
# Se crea una columna nueva que sume el total de las personas afectadas, heridas y fallecidas
df0['total'] = df0.loc[df0['heridos'] >= 0,['heridos','desaparecidos','fallecidos']].sum(axis=1)

df0['total'].fillna(0, inplace=True)
df0

PERSONAS_INVOLUCRADAS     DEPARTAMENTO  heridos  desaparecidos  fallecidos  \
0                     NO         AMAZONAS        0              0           0   
1                     NO        ANTIOQUIA        0              0           0   
2                     NO           ARAUCA        0              0           0   
3                     NO        ATLANTICO        0              0           0   
4                     NO          BOLIVAR        0              0           0   
..                   ...              ...      ...            ...         ...   
59                    SI            SUCRE        2              0           2   
60                    SI           TOLIMA      190             13          52   
61                    SI  VALLE DEL CAUCA      135              5          33   
62                    SI           VAUPES        0              0           8   
63                    SI          VICHADA        0              0           2   

    eventos  total  
0         6      0  
1       746      0  
2       137      0  
3       145      0  
4       215      0  
..      ...    ...  
59        3      4  
60       75    255  
61       41    173  
62        6      8  
63        2      2  

[64 rows x 7 columns]

In [356]:
# Se filtra para que las personas involucradas sean SI
df0 = df0[df0['PERSONAS_INVOLUCRADAS'] != 'NO']
df0=df0.sort_values('total', ascending = False).head(7)
df0

PERSONAS_INVOLUCRADAS  DEPARTAMENTO  heridos  desaparecidos  fallecidos  \
45                    SI  CUNDINAMARCA      899              7         139   
33                    SI     ANTIOQUIA      436             21         155   
41                    SI         CAUCA      193              3          85   
37                    SI        BOYACA      203              7          55   
60                    SI        TOLIMA      190             13          52   
56                    SI     RISARALDA      199              1          24   
51                    SI          META      163              3          28   

    eventos  total  
45      184   1045  
33      135    612  
41       45    281  
37       56    265  
60       75    255  
56       70    224  
51       30    194

In [357]:
# Grafica
df = px.data.tips()
fig = px.sunburst(df0, path=['PERSONAS_INVOLUCRADAS','DEPARTAMENTO'], values='eventos',color='total',hover_data=['heridos','desaparecidos','fallecidos'])
fig.show()

**Interpretación:** Se identifican los 7 departamentos en donde hay mayor cantidad de personas involucradas ya sean heridos, fallecidos y desaparecidos en los desastres naturales o eventos que se presentaron. En primer lugar se encuentra el departamento de Cundinamarca, en el cual ocurrieron 184 eventos que dejaron 899 heridos, 7 desaparecidos y 139 fallecidos para un total de 1045 personas involucradas. El segundo departamento que se destaca en este top 7 es Antioquia, allí ocurrieron 135 eventos que dejaron 436 heridos, 21 desaparecidos y 155 fallecidos para un total de 612 personas involucradas. También están otros departamentos pero en menor cantidad a comparación de Cundinamarca y Antioquia, estos son Tolima, Risaralda, Boyacá, Cauca y Meta.

<p><a name="s3"></a></p>

#**7. ¿Qué más seria interesante ver en este análisis descriptivo?**



###**- Zonas de los departamentos de Chocó, La Guajira y Bolivar que fueron afectadas por inundaciones**

In [334]:
# Instalar paquete
!pip install geojson

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [335]:
# Crear base de datos filtrando por inundación y Chocó, La Guajira, Bolivar
df0=base_datos[['TIPO_DE_EVENTO','DEPARTAMENTO','AFECTADOS','LONGITUD','LATITUD']]
df0 = df0[df0['TIPO_DE_EVENTO']=='INUNDACION'] # Filtrar DataFrame
df0 = df0[df0['DEPARTAMENTO'].isin(['CHOCO','LA GUAJIRA','BOLIVAR'])] # Filtrar DataFrame

In [336]:
token_map = "pk.eyJ1IjoibGF1cmF5YXJjZTE5IiwiYSI6ImNsN2k5eTJuNDBtYjQzb252dnpid3hkOGsifQ.4DM4nHHs4U5PEFlsADNrPQ"
px.set_mapbox_access_token(token_map)

In [337]:
# Grafica
px.scatter_mapbox(df0, lat = 'LATITUD', lon = 'LONGITUD', color =('DEPARTAMENTO'), mapbox_style  = 'streets',
                  color_continuous_scale = px.colors.cyclical.IceFire, size_max = 30, zoom = 4)

**Interpretación:** Como se ha analizado anteriormente, Chocó, La Guajira y Bolivar que son los 3 departamentos con mayor cantidad de personas afectadas, estan ubicados en zonas muy marginadas y desprotegidas en cuanto a los desastres naturales que han ocurrido en Colombia. Graficamente se ubican los lugares en los que han ocurrido inundaciones en estos departamentos.

###**- Zonas con el mayor número de afectados por departamento con el Nivel del Incidente Natural tipo URGENTE**

In [339]:
# Crear base de datos filtrando por el nivel del incidente tipo urgente, se agrupa por departamente y nivel

df0=base_datos[['DEPARTAMENTO','NIVEL_DEL_INCIDENTE_NATURAL','LONGITUD','LATITUD','AFECTADOS']]
df0 = df0[df0['NIVEL_DEL_INCIDENTE_NATURAL']=='URGENTE'] # Filtrar DataFrame

df0=df0.groupby(['DEPARTAMENTO','NIVEL_DEL_INCIDENTE_NATURAL']).agg(
    longitud = ('LONGITUD','mean'),
    latitud = ('LATITUD','mean'),
    afectados = ('AFECTADOS','sum')
    ).reset_index().sort_values('afectados',ascending=False)
df0

DEPARTAMENTO NIVEL_DEL_INCIDENTE_NATURAL   longitud    latitud  \
6          LA GUAJIRA                     URGENTE -72.493174  11.490532   
7           MAGDALENA                     URGENTE -73.884672  10.790636   
3               CHOCO                     URGENTE -77.114267   5.831414   
1             BOLIVAR                     URGENTE -74.894456   9.540048   
8  NORTE DE SANTANDER                     URGENTE -72.799259   8.713507   
0           ANTIOQUIA                     URGENTE -76.695048   6.453351   
4             GUAINIA                     URGENTE -68.008347   3.516363   
5            GUAVIARE                     URGENTE -71.919170   2.484286   
2               CAUCA                     URGENTE -77.491316   2.699475   

   afectados  
6      57758  
7      27308  
3      25000  
1      20690  
8      20300  
0      15000  
4      13757  
5      10305  
2      10264

In [340]:
# Grafica
fig = px.scatter_mapbox(df0, lat='latitud', lon='longitud', color='DEPARTAMENTO', size='afectados',
                  color_continuous_scale=px.colors.cyclical.IceFire, size_max=15, zoom=10,
                  mapbox_style='carto-positron')
fig.show()

**Interpretación:** Para los eventos que han tenido un nivel de clasificación urgente, se grafican los departamentos en los cuales hay afectados. Primero se encuentra nuevamente La Guajira con 57758 afectados, Magdalena con 23087, Chocó con 25000 y Bolivar con 20680 afectados. Se identifica la presencia de los 3 departamentos (Chocó, La Guajira, Bolivar) que ya se ha evidenciado anteriormente en este estudio que tienen más afectados, han ocurrido más desastres naturales y que además los eventos que ocurrieron son de clasificación urgente.

# **Descargar base de datos**

In [ ]:
base_datos.to_csv('/content/drive/MyDrive/base_datos.csv', index = False) # Guardamos el archivo como base_datos.csv
files.download('/content/drive/MyDrive/base_datos.csv') # Descargamos el archivo en nuestra maquina local

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>